### PR TITLE
GEM004 - Constraints on RepresentationIdentifier

### PR DESCRIPTION
**Pull request by Ghesselink**
Creates an csv from the building smart documentation and checks if value for RepresentationIdentifier in the given IFC is valid (i.e. is mentioned in the documentation).

### CODE REVIEW

**Review by aothms — CHANGES_REQUESTED** [review]
Sorry for my comments getting lost:

Summary:
- These are two tables for two attributes
- I'd really think it's better if we refer to a .csv filename in the gherkin statement so it's less magical
- I'd also keep everything out of the repo besides the .csv

**Comment by Ghesselink** [comment]
Latest commit: 

- Only table for RepresentationIdentifer is used (not RepresentationType)
- .csv in gherkin statement
- update description
- Only keep .csv in repository

File: features/steps/steps.py [hunk 1]
```diff
@@ -485,6 +511,44 @@ def step_impl(context, representation_id):
...
+
+        if constraint[-5:] == ".csv'":
+            csv_name = constraint.strip("'")
+            for i, values in enumerate(instances):
+                if not values:
+                    continue
+                attribute = getattr(context, 'attribute', None)
+
+                dirname = os.path.dirname(__file__)
+                filename = Path(dirname).parent / f"resources/{csv_name}"
```
Comment (aothms) [inline features/steps/steps.py:542]: This is a bit inconsistent. If you use Path() with the / operator then do it consistently:

```suggestion
                filename = Path(dirname).parent / "resources" / csv_name
```

**Review by aothms — CHANGES_REQUESTED** [review]

File: features/steps/steps.py (hunk 1 above)
Comment (Ghesselink) [reply features/steps/steps.py:542]: Ah, I did not see this was possible, thanks.

**Review by aothms — APPROVED** [review]

**Pull request merged**